### PR TITLE
DOC: update the .agg doc-string with examples

### PR DESCRIPTION
**Pull request by jreback**
consolidates .agg docs across frame/series & groupby & window

### CODE REVIEW

**Comment by jreback** [comment]
@jorisvandenbossche

File: pandas/core/groupby.py [hunk 1]
```diff
@@ -2749,57 +2750,47 @@ def _selection_name(self):
...
+    >>> s.groupby([1, 1, 2, 2]).agg(['min', 'max'])
+       min  max
+    1    1    2
+    2    3    4
+
+    See also
+    --------
+    pandas.Series.apply
+    pandas.Series.aggregate
+    pandas.Series.transform
```
Comment (jorisvandenbossche) [inline pandas/core/groupby.py]: Shouldn't the see also here refer to groupby().transform as well?

**Review by jorisvandenbossche — APPROVED** [review]
Don't have much time now, but from a quick glance looks good! Nice to have the examples!

**Comment by jreback** [comment]
http://pandas-docs.github.io/pandas-docs-travis/generated/pandas.core.groupby.DataFrameGroupBy.agg.html?highlight=agg#pandas.core.groupby.DataFrameGroupBy.agg

I think some of my See also links are not working :<

**Pull request merged**